core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -851,6 +902,11 @@ protected Builder buildAggregate(Aggregate e, Builder builder,
    * @return true if selectList is required to be added in sqlNode
    */
   boolean isStarInAggregateRel(Aggregate e) {
+
+    PivotRelTrait relTrait = e.getTraitSet().getTrait(PivotRelTraitDef.instance);
+    if (relTrait != null && relTrait.getsubqueryPresentFlag()) {
+      return true;
+    }
```

Comment:
change the variable name and right hand side expression and condition :

```
Optional<PivotRelTrait> pivotRelTrait = 
        Optional.ofNullable(e.getTraitSet().getTrait(PivotRelTraitDef.instance));
    if (pivotRelTrait.isPresent() && pivotRelTrait.get().hasSubquery()) {
      return true;
    }
```

Reply:
resolved